infrastructure/persistence/database/sqlite_novel_repository.py

```
@@ -94,38 +93,35 @@ def save(self, novel: Novel) -> None:
+
+        max_retries = 15
+        base_delay = 1.0
+
+        for attempt in range(max_retries):
+            try:
+                self.db.execute(sql, params)
+                self.db.get_connection().commit()
+                return
+            except sqlite3.OperationalError as e:
+                if "database is locked" in str(e) and attempt < max_retries - 1:
+                    wait = base_delay * (2 ** attempt) + random.uniform(0, 1.0)
```

Ruff (0.15.10)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)